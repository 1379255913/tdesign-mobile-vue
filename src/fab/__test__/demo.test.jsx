/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import displayVue from '@/fab/demos/display.vue';
import mobileVue from '@/fab/demos/mobile.vue';
import textVue from '@/fab/demos/text.vue';

const mapper = {
  displayVue,
  mobileVue,
  textVue,
};

describe('Fab', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Fab ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
