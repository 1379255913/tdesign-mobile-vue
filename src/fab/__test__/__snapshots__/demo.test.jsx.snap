// Vitest Snapshot v1

exports[`Fab > Fab displayVue demo works fine 1`] = `
<div>
  <button
    aria-disabled="false"
    class="t-button t-button--base t-button--primary t-button--shape-round t-fab t-fab--icononly"
    role="button"
    style="right: 16px; bottom: 32px;"
  >
    <!---->
    <span
      class="t-button__text"
    >
      
      <svg
        class="t-icon t-icon-add"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <!--v-if-->
      
    </span>
  </button>
</div>
`;

exports[`Fab > Fab mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Fab 悬浮按钮
  </h1>
  <p
    class="summary"
  >
    当功能使用图标即可表示清楚时，可使用纯图标悬浮按钮，例如：添加、发布。
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      01 类型
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      纯图标悬浮按钮
    </p>
    
    <div
      class="toast-demo"
    >
      <button
        aria-disabled="false"
        class="t-button t-size-l t-button--outline t-button--default t-button--shape-rectangle t-is-block"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          基础使用
          
        </span>
      </button>
      <button
        aria-disabled="false"
        class="t-button t-size-l t-button--outline t-button--default t-button--shape-rectangle t-is-block"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          进阶使用
          
        </span>
      </button>
    </div>
    
  </div>
  <div>
    <button
      aria-disabled="false"
      class="t-button t-button--base t-button--primary t-button--shape-round t-fab t-fab--icononly"
      role="button"
      style="right: 16px; bottom: 32px;"
    >
      <!---->
      <span
        class="t-button__text"
      >
        
        <svg
          class="t-icon t-icon-add"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <!--v-if-->
        
      </span>
    </button>
  </div>
  <!--v-if-->
</div>
`;

exports[`Fab > Fab textVue demo works fine 1`] = `
<div>
  <button
    aria-disabled="false"
    class="t-button t-button--outline t-button--primary t-button--shape-round t-fab"
    role="button"
    style="right: 16px; bottom: 96px;"
  >
    <!---->
    <span
      class="t-button__text"
    >
      
      <svg
        class="t-icon t-icon-call"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M14 11.06c0 .6-.16 1.24-.52 1.76l-.12.16c-.24.35-.58.6-.98.75-.28.1-.46.14-.91.2-1.95.17-4.08-.74-6.18-2.6A10.2 10.2 0 012.32 6.9 6.45 6.45 0 012 5c0-.7.16-1.32.48-1.78.3-.42.88-.76 1.75-1.1a1 1 0 011.17.37l1.67 2.35a1 1 0 01-.1 1.27l-.21.22-.26.25c-.07.06-.17.2-.17.2-.02.3.3.93 1.26 1.9l.28.27c.88.81 1.12.88 1.39.69l.09-.07.87-.62a1 1 0 011.04-.06l.24.13c1.66.89 2.42 1.42 2.5 1.97v.08zm-1 .01c0 .09-.11-.04-.36-.22-.35-.25-.89-.57-1.61-.96l-.24-.12-.79.54-.02.01c-.86.68-1.6.49-3.1-.93-1.14-1.18-1.6-2.05-1.55-2.7.03-.35.27-.67.5-.86l.3-.29.12-.13-1.67-2.36c-.7.27-1.13.53-1.29.74C3.11 4.06 3 4.47 3 5c0 .48.1 1.03.28 1.61a9.2 9.2 0 002.67 3.98c1.91 1.7 3.79 2.5 5.41 2.35l.26-.04c.18-.02.28-.05.4-.1.23-.08.4-.2.54-.4.27-.33.4-.75.44-1.15v-.17z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <span
        class="t-fab__text"
      >
        获取手机号
      </span>
      
    </span>
  </button>
</div>
`;
